modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/application/publisher/test/APIApplicationPublisherTest.java

```diff
@@ -107,6 +107,39 @@ public void testPublish() throws Exception {
 		Assert.assertEquals(0, _serviceTracker.size());
 	}
 
+	@Test
+	public void testPublishUsingAPI() throws Exception {
```

Comment:
Maybe this test should be included in a different Test class, for example ApplicationResourceTest? @LuismiBarcos  what do you think?

Reply:
I write this test due to the use case that we are testing which is publishing an APIApplication. Although, I could move to the APIApplicationModelListener test class

Reply:
Well, I don't know to which file belongs this test.

I will not put it in this class because this class is related to the publisher, which only should receive headless builder's internal ApiApplications.

As I saw on the changes, the code that makes this happen is on the model listeners, so for me, we could move it there.

About having a new class called `ApplicationResourceTest`... I think we don't have that class in the code of the headless-builder and if there is no coding class then no test class should exist. This is how we work on Liferay AFAIK.

Reply:
Great, thanks @LuismiBarcos!! I will do it